delivery/services/organise_service.py

```diff
@@ -207,3 +291,198 @@ def organise_sample_file(self, sample_file, organised_sample_path, lanes):
+            #  values based on variable keys.
+            variables_dict[key] = OrganiseService.sub_values_with_dict(
+                variables_dict[key], 
+                variables_dict)
+
+        for file_element in config["files_to_organise"]:
+            # for each source-destination pair, 
+            # under "files_to_organise", given in the configuration file
+            config_as_list = config_as_list + OrganiseService.process_file_to_organise(
+                file_element, 
+                variables_dict)
+
+        return config_as_list
+        
+    @staticmethod
+    def process_file_to_organise(file_element, variables_dict):
+        """
+        Function to handle a separate element of "files_to_organise"
+          from the cofiguration file.
+        Each element has one source, one destination and options available. 
+        This function will resolve the source and destination into 
+        several source files and destination files if applicable. 
+        Unknowns in the source and destination path will be 
+        resolved by substitutions or by calling resolve_config_regexp().
+        :param file_element: A dictionary with keys: source, destination, options 
+        corresponding to one element of the 
+        "files_to_organise" section in the configuration file.
+        :param variables_dict: A dictionary corresponding to the 
+        section "variables" from the configuration file.
+        :return: A list with all selected files under source. 
+        Each element is a tuple
+          (/source/path/, /destination/path/, {"options": "as dict"}) 
+        """
+        file_element["source"] = OrganiseService.sub_values_with_dict(
+            file_element["source"], 
+            variables_dict)
+        file_element["destination"] = OrganiseService.sub_values_with_dict(
+            file_element["destination"],
+            variables_dict)
+        file_to_organise_list = []
+
+        #Get at list from the filesystem with all files under "source"
+        list_of_mock_files = OrganiseService.get_mock_file_list()
+
+        #Get the filter pattern given in the configuration file and
+        #  substitute available values. 
+        #If filter isn't defined in the configuration file, use an 
+        # empty string as pattern to return all files under source.
+        filter_pattern = OrganiseService.sub_values_with_dict(
+            file_element["options"].get("filter", ""), 
+            variables_dict)
+            
+        source_list = OrganiseService.resolve_config_regexp(
+            filter_pattern, 
+            list_of_mock_files)
+        for source_file in source_list:
+            #substitute unknowns in destination path based on 
+            # varibales found in source file
+            destination_file = OrganiseService.sub_values_with_dict(
+                file_element["destination"], 
+                source_file)
+            file_to_organise_list.append(
```

Comment:
the way "my" function works, it expects the destination to be a full path, i.e. including filename (if it is a file). In other words, you would need to append `os.path.basename(source_file)` to `destination_file`. Another option is that my function takes care of that, which would be closer to how e.g. `cp` and `ln` works on linux, but I assumed my function would be as dumb as possible.. we can have a quick discussion about this.